pkg/cluster/deploybaseresources.go

```diff
@@ -444,6 +454,60 @@ func (m *manager) setMasterSubnetPolicies(ctx context.Context) error {
 	return err
 }
 
+func (m *manager) generateFederatedIdentityCredentials(ctx context.Context) error {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return nil
+	}
+
+	if m.doc.OpenShiftCluster.Properties.ClusterProfile.OIDCIssuer == nil {
+		return fmt.Errorf("OIDCIssuer is nil for the cluster with ID: %s", m.doc.OpenShiftCluster.ID)
+	}
+
+	Issuer := to.StringPtr((string)(*m.doc.OpenShiftCluster.Properties.ClusterProfile.OIDCIssuer))
+
+	platformWIRolesByRoleName := m.platformWorkloadIdentityRolesByVersion.GetPlatformWorkloadIdentityRolesByRoleName()
+	platformWorkloadIdentities := m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities
+
+	for _, identity := range platformWorkloadIdentities {
+		identityResourceId, err := azure.ParseResourceID(identity.ResourceID)
+		if err != nil {
+			return err
+		}
+
+		platformWIRole, exists := platformWIRolesByRoleName[identity.OperatorName]
+		if !exists {
+			return fmt.Errorf("platform workload identity role %s not found", identity.OperatorName)
+		}
+
+		federatedIdentityCredentialResourceName, err := m.getPlatformWorkloadIdentityFedertedCredName(identity)
+		if err != nil {
+			return err
+		}
+
+		for _, sa := range platformWIRole.ServiceAccounts {
+			_, err := m.clusterMsiFederatedIdentityCredentials.CreateOrUpdate(
+				ctx,
+				identityResourceId.ResourceGroup,
+				identityResourceId.ResourceName,
+				federatedIdentityCredentialResourceName,
+				armmsi.FederatedIdentityCredential{
+					Properties: &armmsi.FederatedIdentityCredentialProperties{
```

Comment:
See this as an example for setting properties: https://azure.github.io/azure-workload-identity/docs/quick-start.html#4-create-an-aad-application-or-user-assigned-managed-identity-and-grant-permissions-to-access-the-secret

Reply:
Anything specific you want me to look for? If you're talking about the way we set `Audiences: openshift`,[ here is the slack conversation details](https://redhat-external.slack.com/archives/C06FRQKUMFF/p1723242301968319?thread_ts=1723242231.734469&cid=C06FRQKUMFF) why we set it to `openshift` instaed of `"api://AzureADTokenExchange"`